de.metas.vertical.pharma/src/main/java/de/metas/impexp/bpartner/IFABPartnerLocationImportHelper.java

```diff
@@ -92,47 +96,50 @@ private I_C_BPartner_Location fetchAndUpdateExistingBPLocation(@NonNull final I_
 	{
 		final List<I_C_BPartner_Location> existentBPLocations = Services.get(IBPartnerDAO.class).retrieveBPartnerLocations(importRecord.getC_BPartner());
-		if (!existentBPLocations.isEmpty())
+		if (existentBPLocations.isEmpty())
 		{
-			final Address importAddress = Address.builder()
-					.counrytId(importRecord.getC_Country_ID())
-					.city(importRecord.getb00ortzu())
-					.address1(buildAddress1(importRecord))
-					.address2(buildAddress2(importRecord))
-					.postal(importRecord.getb00plzzu1())
-					.pobox(buildPOBox(importRecord))
-					.build();
-
-			final List<Address> matchedAddreses = existentBPLocations.stream()
-					.map(bpLocation -> Address.builder()
-							.counrytId(importRecord.getC_Country_ID())
-							.city(importRecord.getb00ortzu())
-							.address1(buildAddress1(importRecord))
-							.address2(buildAddress2(importRecord))
-							.postal(importRecord.getb00plzzu1())
-							.pobox(buildPOBox(importRecord))
-							.bpLocationId(bpLocation.getC_BPartner_Location_ID())
-							.build())
-					.collect(ImmutableList.toImmutableList());
-
-			final List<Address> filtered = matchedAddreses.stream()
-					.filter(address -> isMatched(importAddress, address))
-					.collect(ImmutableList.toImmutableList());
-
-			if (filtered.size() > 0)
-			{
-				bpartnerLocation = Services.get(IBPartnerDAO.class).getBPartnerLocationById(BPartnerLocationId.ofRepoId(importRecord.getC_BPartner_ID(), matchedAddreses.get(0).getBpLocationId()));
-				updateExistingBPartnerLocation(importRecord, bpartnerLocation);
-			}
-
+			return null;
 		}
-		return bpartnerLocation;
+		
+		final Address importAddress = Address.builder()
+				.counrytId(importRecord.getC_Country_ID())
+				.city(importRecord.getb00ortzu())
+				.address1(buildAddress1(importRecord))
+				.address2(buildAddress2(importRecord))
+				.postal(importRecord.getb00plzzu1())
+				.pobox(buildPOBox(importRecord))
+				.build();
+
+		final List<Address> matchedAddreses = existentBPLocations.stream()
+				.map(bpLocation -> Address.builder()
+						.counrytId(importRecord.getC_Country_ID())
+						.city(importRecord.getb00ortzu())
+						.address1(buildAddress1(importRecord))
+						.address2(buildAddress2(importRecord))
+						.postal(importRecord.getb00plzzu1())
+						.pobox(buildPOBox(importRecord))
+						.bpLocationId(bpLocation.getC_BPartner_Location_ID())
+						.build())
+				.collect(ImmutableList.toImmutableList());
+
+		final List<Address> filtered = matchedAddreses.stream()
+				.filter(address -> isMatched(importAddress, address))
+				.collect(ImmutableList.toImmutableList());
+
+		if (filtered.size() > 0)
+		{
+			final I_C_BPartner_Location bpartnerLocation = Services.get(IBPartnerDAO.class).getBPartnerLocationById(BPartnerLocationId.ofRepoId(importRecord.getC_BPartner_ID(), matchedAddreses.get(0).getBpLocationId()));
+			updateExistingBPartnerLocation(importRecord, bpartnerLocation);
```

Comment:
don't we want to return the updated `bpartnerLocation` here?

Reply:
Yes! You are right!